pkg/provenance/provenance.go

```diff
@@ -139,26 +157,67 @@ func NewSpec() *Spec {
 
 func FindProvenanceObjectByName(name string, allObjects []ProvenanceOfObject) *ProvenanceOfObject {
 	for _, value := range allObjects {
+		print(value.Name)
 		if name == value.Name {
 			return &value
 		}
 	}
 	return nil
 }
-
+// This String function must return the same output upon
+// different calls. So I am doing ordering and sorting
+// here because map is unordered and gives random outputs
+// when printing
 func (s *Spec) String() string {
 	var b strings.Builder
 	fmt.Fprintf(&b, "Version: %d\n", s.Version)
-	for attribute, data := range s.AttributeToData {
-		fmt.Fprintf(&b, "%s: %s\n", attribute, data)
+
+	var keys []string
+	for k, _ := range s.AttributeToData {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	for _, attribute := range keys {
+		data := s.AttributeToData[attribute]
+		integer, ok := data.(int)
```

Comment:
Can the data be only of int and list of map[string]string?

Reply:
So I only check those two because my reasoning was that the string conversion of a map will be in random ordering, and converting an int to string gives problems the way I did it so I check it here. The default string conversion of slices and strings are working fine